runtime/jcl/common/thread.cpp

```diff
@@ -603,28 +603,94 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+/**
+ * Stack walk function to find the most recent scoped value binding in the stack.
+ *
+ * @param currentThread the current thread
+ * @param walkState the stack walk state
+ */
+static UDATA
+findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->userData1) {
+		rc = J9_STACKWALK_STOP_ITERATING;
+	} else if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
+
+		if (((J9VMJAVALANGTHREAD_OR_NULL(vm) == methodClass)
+			|| (J9VMJAVALANGVIRTUALTHREAD_OR_NULL(vm) == methodClass)
+			|| (J9VMJDKINCUBATORCONCURRENTSCOPEDVALUECARRIER_OR_NULL(vm) == methodClass))
+			&& J9UTF8_LITERAL_EQUALS(J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodName), "runWith")
+		) {
+			walkState->flags |= J9_STACKWALK_ITERATE_O_SLOTS;
```

Comment:
Add a comment here to highlight the issue and justify the approach taken. Something like ...

The goal is to find the first argument of the runWith method, which is an instance of jdk.incubator.concurrent.ScopedValue$Snapshot. But the first argument can't be reliably retrieved for any compiled method. The runWith method has two arguments, and there is only one argument which is an instance of jdk.incubator.concurrent.ScopedValue$Snapshot. So, all the arguments are looked at to find the Snapshot instance.